modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/EncodedRosettaTransaction.scala

```diff
@@ -0,0 +1,32 @@
+package org.tessellation.rosetta.domain
+
+import java.nio.charset.{Charset, StandardCharsets}
+import java.util.Base64
+
+import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
+import org.tessellation.schema.transaction.DAGTransaction
+
+import derevo.circe.magnolia.{customizableDecoder, customizableEncoder}
+import derevo.derive
+import io.circe.parser._
+import io.circe.syntax._
+
+object transaction {
+  @derive(customizableDecoder, customizableEncoder)
+  final case class EncodedRosettaTransaction(value: String)
+
+  object EncodedRosettaTransaction {
```

Comment:
you can prepare `Base64Codec` in the same way as the `BinaryCodec` is created. it doesn't need to be transaction-specific

Reply:
Is this what you are referring to?

https://github.com/Constellation-Labs/tessellation/blob/37466d8b50bbb69a4010df3b293c0664bee9fb4f/modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/consensus/types/ConsensusHealthStatus.scala#L19-L25

Reply:
No, this is the usage of the BinaryCodec. Please take a look into the implementation